src/pseudopeople/utilities.py

```diff
@@ -7,22 +7,24 @@
 from vivarium.framework.randomness import RandomnessStream
 from vivarium.framework.randomness.index_map import IndexMap
 
-from pseudopeople.constants import metadata, paths
+from pseudopeople.constants import metadata
 
 
-def get_randomness_stream(dataset_name: str, seed: int) -> RandomnessStream:
+def get_randomness_stream(dataset_name: str, seed: int, index: pd.Index) -> RandomnessStream:
     return RandomnessStream(
-        dataset_name, lambda: pd.Timestamp("2020-04-01"), seed, IndexMap()
+        key=dataset_name,
+        clock=lambda: pd.Timestamp("2020-04-01"),
+        seed=seed,
+        index_map=IndexMap(size=max(index) + 1),
```

Comment:
Consider having a minimum map size of 1 million and using the 2x the max(index) rather than 1 +